src/fetchProjectItems.ts

```diff
@@ -0,0 +1,96 @@
+import { GetProjectsDocument, GetProjectsQuery, GetProjectsQueryVariables } from "./generated/graphql-operations";
+import { ProjectDetails, ProjectItem } from "./types";
+import { print } from "graphql/index";
+import { octokit } from "./octokit";
+
+export async function fetchProjectItems(
+  params: GetProjectsQueryVariables,
+  recursiveCollection?: ProjectDetails
+): Promise<ProjectDetails> {
+  console.log(`Fetching projects`, params)
+  // @ts-ignore
+  const project: GetProjectsQuery = await fetchProject(params).catch(console.error)
+
+  console.log(project)
+
+  if (project.organization?.projectV2?.items?.edges) {
+    const { hasNextPage, endCursor } = project.organization?.projectV2?.items?.pageInfo
+    const items: ProjectItem[] = [];
+
+    project.organization?.projectV2?.items?.edges
+      ?.map(item => item?.node)
+      ?.map(item => {
+        if (item) {
+          if (item.content && Object.keys(item.content).length > 0 && item.content.__typename === "Issue") {
+            let projectItem: ProjectItem = {
+              id: item.id,
+              content: {
+                ...item.content,
+                customFields: {}
+              },
+            }
+
+            if (item.fieldValues) {
+              item.fieldValues.edges
+                ?.map(fv => {
+                  if (!fv?.node) return;
+
+                  switch(fv.node.__typename) {
+                    case "ProjectV2ItemFieldLabelValue":
+                      projectItem.content.labels = fv.node.labels?.nodes
+                        ?.filter(label => !!label)
+                        ?.map(label => {
+                          return {
+                            name: label!.name,
+                            url: label!.url,
+                            color: label!.color,
+                          }
+                        })
+                      break;
+                    case "ProjectV2ItemFieldSingleSelectValue":
+                      if (fv.node?.field.__typename === "ProjectV2SingleSelectField") {
+                        projectItem.content.customFields[fv.node.field.name] = fv.node.name
+                      }
+                      break
+                    case "ProjectV2ItemFieldNumberValue":
+                      if (fv.node?.field.__typename === "ProjectV2Field") {
+                        projectItem.content.customFields[fv.node.field.name] = fv.node.number
+                      }
+                      break
+                  }
+                })
+            }
```

Comment:
I can't understand what's going on here. Could you simplify/split/comment it somehow?

Reply:
I see why that looks complicated
I tried to simplify a bit, added comments what exactly going on and reasoning and moved out the label extraction.
However It was harder to split/simplify the rest of the fields extractions due to the fact of GraphQL + Typings, generate custom type accordingly to what i have queried, and it doesn't unfortunately split its nested types, so I could use these NestedTypeName in my separate method. The only way is to copy some part of generated type into a argument type definition but it's toooo long and really complicates further support. 
As if you add more fields, you'll have to add them in these definitions as well
<img width="2058" alt="image" src="https://user-images.githubusercontent.com/1177472/200603536-afda02c3-05b9-4bd7-b708-d568cb094857.png">
like this 
<img width="705" alt="image" src="https://user-images.githubusercontent.com/1177472/200603727-c123ca6d-870a-4f59-9fae-06aedd5be6c3.png">

I know it's not ideal, but i hope comments will simplify support in future

Reply:
Comments do help, thank you